pkg/encoding/lorawan/messages.go

```diff
@@ -639,3 +639,30 @@ func GetUplinkMessageIdentifiers(phyPayload []byte) (ttnpb.EndDeviceIdentifiers,
 		return ttnpb.EndDeviceIdentifiers{}, errUnknown("MType")(mhdr.MType.String())
 	}
 }
+
+// GetUplinkMessageIdentifiersFromMessage retrieves the EndDeviceIdentifers from a LoRaWAN message (except DeviceID).
+func GetUplinkMessageIdentifiersFromMessage(msg *ttnpb.Message) (ttnpb.EndDeviceIdentifiers, error) {
+	var ids ttnpb.EndDeviceIdentifiers
+	if msg == nil {
+		return ttnpb.EndDeviceIdentifiers{}, errInvalidPayload.New()
+	}
+	switch msg.MHDR.MType {
+	case ttnpb.MType_CONFIRMED_UP, ttnpb.MType_UNCONFIRMED_UP:
+		payload := msg.GetMACPayload()
```

Comment:
Is it guaranteed that `payload != nil` if the `MType` is set?

Reply:
I would suggest something like;

```go
if macPayload := msg.GetMACPayload(); payload != nil {
   ...
} else if ...; ... != nil {
   ...
}
```

Reply:
Changed to only access the payload if it is not nil. Is the mtype check useful?

Reply:
No